lib/middleware/keep_request.ex

```diff
@@ -0,0 +1,13 @@
+defmodule PINXS.Middleware.KeepRequest do
+  @behaviour Tesla.Middleware
+  alias Tesla.Env
```

Comment:
Might be worth a comment about what this middleware is doing and why.

Reply:
I'll cherry pick the commit I did for V3 as I did add it there